tests/output.rs

```diff
@@ -1,228 +1,118 @@
 use assert_cmd::prelude::*;
-use once_cell::sync::Lazy;
-use predicates::{prelude::*, BoxPredicate};
-use std::{
-    env,
-    ffi::OsStr,
-    path::{Path, PathBuf},
-    process::Command,
-};
+use predicates::prelude::*;
+use std::{env, path::PathBuf, process::Command};
 
-struct Location {
-    line: usize,
-    #[allow(dead_code)]
-    column: usize,
+#[macro_export]
+macro_rules! regex {
+    ($re:literal $(,)?) => {{
+        static RE: once_cell::sync::OnceCell<regex::Regex> = once_cell::sync::OnceCell::new();
+        RE.get_or_init(|| regex::Regex::new($re).unwrap())
+    }};
 }
 
-impl From<(usize, usize)> for Location {
-    fn from(value: (usize, usize)) -> Self {
-        Self {
-            line: value.0,
-            column: value.1,
-        }
-    }
-}
-
-struct Range {
-    pub start: Location,
-    #[allow(dead_code)]
-    pub end: Location,
+fn get_fixture_dir_path_from_name(fixture_dir_name: &str) -> PathBuf {
+    // per https://andrewra.dev/2019/03/01/testing-in-rust-temporary-files/
+    let root_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let mut path: PathBuf = root_dir.into();
+    path.push("tests/fixtures");
+    path.push(fixture_dir_name);
+    path
 }
 
-impl From<((usize, usize), (usize, usize))> for Range {
-    fn from(value: ((usize, usize), (usize, usize))) -> Self {
-        Self {
-            start: value.0.into(),
-            end: value.1.into(),
+fn parse_command_and_output(command_and_output: &str) -> CommandAndOutput {
+    let mut lines = command_and_output.split("\n").collect::<Vec<_>>();
+    if !lines.is_empty() {
+        if regex!(r#"^\s*$"#).is_match(&lines[0]) {
```

Comment:
This is fine, but personally I have to think a bit to mentally parse the regex, so I'd probably write:

```rs
if lines[0].trim().is_empty() {
```